crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -76,6 +91,55 @@ fn execute_command<E: fmt::Debug, S: StorageView<E>>(
     command: Command,
 ) -> Result<(), ExecutionError> {
     let results = match command {
+        Command::MakeMoveVec(args) if args.is_empty() => {
+            vec![Value::EmptyVec]
+        }
+        Command::MakeMoveVec(args) => {
+            let mut res = vec![];
+            leb128::write::unsigned(&mut res, args.len() as u64).unwrap();
+            let mut arg_iter = args.into_iter().enumerate();
+            let (idx, first_arg) = arg_iter.next().unwrap();
+            let first_value: Value = context.take_arg(CommandKind::MakeMoveVec, idx, first_arg)?;
+            let mut used_in_non_entry_move_call = first_value.was_used_in_non_entry_move_call();
+            let mut type_ = first_value.type_();
+            res.extend(first_value.to_bcs_bytes());
+            for (idx, v) in arg_iter {
+                let value: Value = context.take_arg(CommandKind::MakeMoveVec, idx, v)?;
+                if let Some(joined) = context.join(type_, value.type_())? {
+                    type_ = joined
+                } else {
+                    panic!("type mismatch")
+                }
+                used_in_non_entry_move_call =
+                    used_in_non_entry_move_call || value.was_used_in_non_entry_move_call();
+                res.extend(value.to_bcs_bytes());
+            }
+            let inner_ty = match type_ {
+                ValueType::AnyPrimitive | ValueType::AnyVec => panic!(
```

Comment:
@amnn I'll probably need your opinion here as you have been following the runtime closely. 

This case is also a bit strange, and with the clone vs take above, I wonder if `MakeMoveVec` should just take a type tag? 
Pro:
- No Value::EmptyVec
- No Weird edge case here
- we can check for primitives first which would simplify the join case for `AnyPrimitive`
Con: 
- Makes it not possible to migrate `CallArg::ObjVec` naively, as the type tag would now need to be specified.

Reply:
I went ahead and added the type argument, which solves a lot of issues. I made it optional for non-empty vector of objects, just to make the migration a tad easier... but hopefully that isn't confusing.

Reply:
+1 to including the type tag at least for the empty vector case -- the behaviour of an empty vector (w.r.t. droppability) changes based on its type, so it makes sense to me that it would need to be specified.  The other option is to have `ObjVec` and `PureVec` (where "pure" implies "copy + drop")?

Reply:
I tried a few different approaches and like where I landed (I think)
- If it is empty, you must specify a type tag
- If the elements are objects, you _don't_ have to specify a type tag
- If the elements are not objects, clane/take is decided by the abilities of the type tag